webhook_to_fedora_messaging/endpoints/parser/parser.py

```diff
@@ -0,0 +1,63 @@
+
+from  flask import Request
+from werkzeug.datastructures.headers import EnvironHeaders
+import hashlib
+import hmac
+from webhook_to_fedora_messaging_messages.github.github import GithubMessageV1
+from ..util import fas_by_github
+
+
+
+def msg_parser(service_type: str, request: Request):
+    if service_type.lower() == "github":
+        return _github_parser(request)
+
+
+def _github_parser(request: Request, secret=True) -> GithubMessageV1:
+    """
+        Convert Flask request objects into desired FedMsg format.
+        
+        Args:
+            headers: Flask request.headers
+            body: A dictionary contains payload
+            secret: Specifies whether the webhook has secret key feature on or not
+    """
+    
+    headers = get_github_headers(request.headers)
+    
+    secret = "00b863c62c96aa67aeea966104362b1136ce883d" # will come from a data source
+    if (secret and verify_signature(request.data, secret, headers['X-Hub-Signature-256'])) or not secret:
```

Comment:
It's usually clearer to handle the error cases first. For example, instead of:
```python
if condition 1:
    if condition 2:
        return do_something()
    return "2 not matched"
return "1 not matched"
```
it is usually clearer to do:
```python
if not condition1:
    return "1 not matched"
if not condition2:
    return "2 not matched"
return do_something()
```
This avoid indenting too much, and puts the response of the error cases closer to the condition.
The reasoning is: deal with the error cases first, and then do your normal path.

As a result, here I would do something like:
```python
if secret and not verify_signature(...):
    return None
topic = ...
agent = ...
return GithubMessageV1(...)
```